### ISSUE TITLE
ANSI C signal handling for gmxapi

### ISSUE THREAD

**Issue by eirrgang** [kassonlab/gmxapi#137]
Currently, GROMACS installs a signal handler that traps things like SIGINT and SIGTERM. Unless receiving several INTs or stronger signals, it sets a stop_condition global state in the library and does not re-raise the signal. The consequence is that ctrl-C does not directly interrupt a gmxapi script, but just interrupts the currently running simulation and then the script proceeds.

I kind of like the granularity of being able to issue a signal to just one element of the workflow, but it brings up a few questions.

- [ ] If we like the idea of manipulating a workflow this way -- overriding the stop condition with a signal during execution -- we should be clear about what defines success and how downstream operations should interpret the output
- [ ] What do we do if the user wants to actually stop the whole script? We could trap and reraise SIGINT with the common pattern that two INTs in a row has special meaning (to the session)
- [ ] To play nice with job scheduling systems, scripts run non-interactively should probably cleanly terminate the entire workflow on the first SIGINT
- [ ] We should also consider appropriate meanings of SIGHUP and SIGTERM

In short, we should explicitly add an ANSI C signal handler to gmxapi, but its behavior warrants some discussion.
